sigstore-java/src/main/java/dev/sigstore/encryption/signers/EcdsaSigner.java

```diff
@@ -43,6 +43,9 @@ public byte[] sign(byte[] artifact)
   @Override
   public byte[] signDigest(byte[] artifactDigest)
       throws NoSuchAlgorithmException, InvalidKeyException, SignatureException {
+    if (artifactDigest.length > 64) {
+      throw new SignatureExeption("Artifact digest cannot be longer than 64 bytes for this mode");
+    }
```

Comment:
How is this better than the exception from `signature.update(artifactDigest);` ?

Reply:
It's better in that it's a controlled exception with a meaningful description attached to it.

The exception thrown is:
```
== Java Exception: java.lang.ArrayIndexOutOfBoundsException: arraycopy: last destination index 65 out of bounds for byte[64]
 at java.base/java.lang.System.arraycopy(Native Method)
 at jdk.crypto.ec/sun.security.ec.ECDSASignature$RawECDSA.engineUpdate(ECDSASignature.java:158)
 at java.base/java.security.Signature$Delegate.engineUpdate(Signature.java:1414)
 at java.base/java.security.Signature.update(Signature.java:903)
 at java.base/java.security.Signature.update(Signature.java:872)
 at dev.sigstore.encryption.signers.EcdsaSigner.signDigest(EcdsaSigner.java:48)
 at fuzzing.SignerVerifierFuzzer.fuzzerTestOneInput(SignerVerifierFuzzer.java:37)
```

Reply:
I wonder if the check should be implemented in the downstream library instead. Of course, it is easier to sweep it under the rug here, however, it is hard to understand why we have an artificial limit of 64 here.

So I would suggest filing an issue/PR for the impl library rather than sigstore-java

Reply:
That makes sense to me. Let's make a PR there and see how it's retrieved.

Reply:
Just checking in? Was there any resolution on this? Is there a PR somewhere to follow up on?

Reply:
Ah, I missed this! Apologies -- @arthurscchan could you help with this one?

Reply:
I just spoke to @arthurscchan and @AdamKorcz who said it may not be the correct choice to change upstream. @AdamKorcz could you elaborate?

Reply:
The best "upstream" place to fix this issue is at the provider level. However, this is not a suitable place to fix this issue, since there are many providers that sigstore-java users can consume and/or write their own providers. As such, sigstore-java is the best place to do the digest length check.